tests/integration/test_iam.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+
+from databricks.sdk.core import DatabricksError
+
+
+def test_filtering_groups(w, random):
+    all = w.groups.list(filter=f'displayName eq any-{random(12)}')
+    found = len(list(all))
+    assert found == 0
+
+
+def test_scim_error_unmarshall(w, random):
+    with pytest.raises(DatabricksError) as exc_info:
+        w.groups.list(filter=random(12))
+    assert 'Given filter operator is not supported' in str(exc_info.value)
+
+
+@pytest.mark.parametrize("path,call", [
+    ("/api/2.0/preview/scim/v2/Users", lambda w: w.users.list(count=1)),
+    ("/api/2.0/preview/scim/v2/Groups", lambda w: w.groups.list(count=1)),
+    ("/api/2.0/preview/scim/v2/ServicePrincipals", lambda w: w.service_principals.list(count=1)),
+])
+def test_workspace_users_list_pagination(w, path, call):
+    raw = w.api_client.do('GET', path)
+    total = raw['totalResults']
+    all = call(w)
+    found = len(list(all))
+    assert found == total
+
+
+@pytest.mark.parametrize("path,call", [
+    # there are ~7k users in our aws prod account
```

Comment:
```suggestion
```
i don't think it should be in public code

Reply:
excellent point